### ISSUE TITLE
Loading specific github page freezes up with 100% CPU usage on one core

### ISSUE THREAD

**Issue by rzetterberg** [The-Compiler/qutebrowser#1179]
# Description

Loading [this page](https://github.com/django/django/blob/master/tests/modeladmin/tests.py) results in a freeze with 100% CPU usage on one core.

To close qutebrowser I have to do a `pkill -9 qutebrowser`.

I'm running without javascript, sessions and cookies (see full config below).

# Environment

```bash
$ qutebrowser --version
qutebrowser v0.3.0
Git commit: 

CPython: 3.4.3
Qt: 5.4.2, runtime: 5.4.2
PyQt: 5.4.2
```

## Config

NB: aliases, searchengines and color is removed since it should not be relevant.

```cfg
[general]
ignore-case = smart
wrap-search = true
startpage = https://www.duckduckgo.com
default-page = ${startpage}
auto-search = naive
auto-save-config = true
auto-save-interval = 15000
editor = gvim -f "{}"
editor-encoding = utf-8
private-browsing = true
developer-extras = false
print-element-backgrounds = true
xss-auditing = false
site-specific-quirks = true
default-encoding =
new-instance-open-target = tab
log-javascript-console = false
save-session = false
session-default-name =
[ui]
zoom-levels = 25%,33%,50%,67%,75%,90%,100%,110%,125%,150%,175%,200%,250%,300%,400%,500%
default-zoom = 100%
downloads-position = north
message-timeout = 2000
message-unfocused = false
confirm-quit = never
display-statusbar-messages = false
zoom-text-only = false
frame-flattening = false
user-stylesheet = ::-webkit-scrollbar { width: 0px; height: 0px; }
css-media-type =
smooth-scrolling = false
remove-finished-downloads = false
hide-statusbar = false
window-title-format = {perc}{title}{title_sep}qutebrowser
hide-mouse-cursor = false
modal-js-dialog = false
[network]
do-not-track = true
accept-language = en-US,en
user-agent =
proxy = system
proxy-dns-requests = true
ssl-strict = ask
dns-prefetch = true
[completion]
auto-open = true
download-path-suggestion = path
timestamp-format = %Y-%m-%d
show = true
height = 50%
cmd-history-max-items = 0
web-history-max-items = 0
quick-complete = true
shrink = false
[input]
timeout = 500
partial-timeout = 1000
insert-mode-on-plugins = false
auto-leave-insert-mode = true
auto-insert-mode = false
forward-unbound-keys = auto
spatial-navigation = false
links-included-in-focus-chain = true
rocker-gestures = false
mouse-zoom-divider = 512
[tabs]
background-tabs = false
select-on-remove = right
new-tab-position = right
new-tab-position-explicit = last
last-close = ignore
hide-auto = false
hide-always = false
wrap = true
movable = true
close-mouse-button = middle
position = north
show-favicons = true
width = 20%
indicator-width = 3
indicator-space = 3
tabs-are-windows = false
title-format = {index}: {title}
mousewheel-tab-switching = true
[storage]
download-directory =
maximum-pages-in-cache =
object-cache-capacities =
offline-storage-default-quota =
offline-web-application-cache-quota =
offline-storage-database = true
offline-web-application-storage = true
local-storage = true
cache-size = 52428800
[content]
allow-images = true
allow-javascript = false
allow-plugins = false
webgl = true
css-regions = true
hyperlink-auditing = false
geolocation = ask
notifications = ask
javascript-can-open-windows = false
javascript-can-close-windows = false
javascript-can-access-clipboard = false
ignore-javascript-prompt = false
ignore-javascript-alert = false
local-content-can-access-remote-urls = false
local-content-can-access-file-urls = true
cookies-accept = never
cookies-store = false
host-block-lists = http://www.malwaredomainlist.com/hostslist/hosts.txt,http://someonewhocares.org/hosts/hosts,http://winhelp2002.mvps.org/hosts.zip,http://malwaredomains.lehigh.edu/files/justdomains.zip,http://pgl.yoyo.org/adservers/serverlist.php?hostformat=hosts&mimetype=plaintext
host-blocking-enabled = true
[hints]
border = 1px solid 
opacity = 0.7
mode = letter
chars = asdfghjkl
min-chars = 1
scatter = true
uppercase = false
auto-follow = true
next-regexes = \bnext\b,\bmore\b,\bnewer\b,\b[>→≫]\b,\b(>>|»)\b,\bcontinue\b
prev-regexes = \bprev(ious)?\b,\bback\b,\bolder\b,\b[<←≪]\b,\b(<<|«)\b
```

**Comment by The-Compiler** [comment]
Qt had a performance issue when rendering large highlighted source code files on GitHub - this has been fixed in Qt 5.5 when I recall correctly. Unfortunately some distributions insist on shipping outdated versions :wink:

By the way: qutebrowser v0.3.0 is rather old - please upgrade!

**Comment by rzetterberg** [comment]
@The-Compiler I see, thanks for the clarification! :+1:
